contentgrid-appserver-query-engine-api/src/main/java/com/contentgrid/appserver/query/engine/api/EventConsumer.java

```diff
@@ -0,0 +1,11 @@
+package com.contentgrid.appserver.query.engine.api;
+
+import com.contentgrid.appserver.application.model.Application;
+import com.contentgrid.appserver.application.model.values.EntityName;
+import com.contentgrid.appserver.query.engine.api.data.EntityData;
+
+public interface EventConsumer {
```

Comment:
I think this is missing handlers for relation linking/unlinking?

We don't have specific *webhooks* for those events (it's just 'update'), but I think we should distinguish them in the infrastructure.

We need to see a bit how to handle the relations exactly, if we have different methods for to-one and to-many relations.

There are 2 differences that may warrant having different functions:
- for to-one relations, when linking a new item, there is an existing relation item that is possibly replaced. The object that is replaced should be reported as well.
- for to-many relations, it's possible to add/remove items in bulk. I don't think we want to dispatch _n_ events in that case; just one that lists all the items that have been added or removed.